python/ray/_private/thirdparty/pyamdsmi/pyamdsmi.py

```
@@ -0,0 +1,552 @@
+        return _rocml_get_function_ptr_cache[name]
+
+    lib_load_lock.acquire()
+    try:
+        # ensure library was loaded
+        if rocm_lib == None:
+            raise ROCMLError_Uninitialized
+        try:
+            _rocml_get_function_ptr_cache[name] = getattr(rocm_lib, name)
+            return _rocml_get_function_ptr_cache[name]
+        except AttributeError:
+            raise ROCMLError_FunctionNotFound
+    finally:
+        # lock is always freed
+        lib_load_lock.release()
+
+
+def _load_rocm_library():
+    """Load ROCm library if not already loaded"""
+    global rocm_lib
+
+    if rocm_lib == None:
+
+        lib_load_lock.acquire()
+
+        try:
+            if rocm_lib == None:
+                try:
+                    if sys.platform[:3] == 'win':
+                        raise ROCMLError_NotSupported('Windows platform is not supported yet')
+                    else:
+                        # assume linux
+                        path_librocm = _find_lib_rocm()
+                        cdll.LoadLibrary(path_librocm)
+                        rocm_lib = CDLL(path_librocm)
+                except OSError:
+                    raise ROCMLError_LibraryNotFound('ROCm library not found')
+                if rocm_lib == None:
+                    raise ROCMLError_LibraryNotFound('ROCm library not found')
+        finally:
+            lib_load_lock.release()
+
+
+def _find_lib_rocm():
+    """search for librocm and returns path
+    if search fails, returns empty string
+    """
+    rocm_path = os.environ.get('ROCM_PATH', '/opt/rocm')
+    rocm_lib_path = join(rocm_path, f'lib/{LIBROCM_NAME}')
+    return rocm_lib_path if isfile(rocm_lib_path) else ''
+
+
+def _driver_initialized():
+    """ Returns true if amdgpu is found in the list of initialized modules
+    """
+    initialized = ''
+    try:
+        initialized = str(subprocess.check_output("cat /sys/module/amdgpu/initstate |grep live", shell=True))
```

Need to add `stderr=subprocess.DEVNULL` to suppress the error log. This caused the test failure.